app/js/sumo_db.js

```diff
@@ -122,6 +123,41 @@
       return Promise.all([vote_promise, solved_promise]);
     },
 
+    remove_tags: function(tags, question_id) {
+      var endpoint = API_V2_BASE + 'question/';
+      endpoint += question_id + '/remove_tags/';
+      endpoint += '?format=json'; // TODO bug 1088014
+
+      return request_with_auth(endpoint, 'POST',
+        {tags: JSON.stringify(tags)});
+    },
+
+    add_tags: function(tags, question_id) {
+      var endpoint = API_V2_BASE + 'question/';
+      endpoint += question_id + '/add_tags/';
+      endpoint += '?format=json'; // TODO bug 1088014
+
+      return request_with_auth((endpoint, 'POST',
+        {tags: JSON.stringify(tags)})).then(JSON.parse);
+    },
+
+    set_topic_for_question: function(topic, question_id) {
+      var promise = SumoDB.get_question(question_id);
+
+      return promise.then(function(question) {
+        var topic_tags = [];
+        for (var i = 0; i < question.tags.length; i++) {
+          var tag = question.tags[i];
+          if (tag.startsWith(TOPIC_PREFIX)) {
+            topic_tags.push(tag);
+          }
+        }
+        return SumoDB.remove_tags(topic_tags, question_id);
+      }).then(function() {
```

Comment:
I think this should happen serially because if we add the tag before we call `get_question`, all the topic tags will be removed and the one we are trying to add will not get added.